sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/_operations/_patch.py

```diff
@@ -0,0 +1,226 @@
+        return self._status
+
+    def finished(self) -> bool:
+        return self.status() in {"finished", "failed"}
+
+    def resource(self):
+        if self._deserialization_callback:
+            return self._deserialization_callback(self._latest_response)
+
+        return self._latest_response
+
+
+class ConfidentialLedgerClientOperationsMixin(GeneratedOperationsMixin):
+    def begin_get_ledger_entry(
+        self, transaction_id: str, *, collection_id: Optional[str] = None, **kwargs: Any
+    ) -> LROPoller[JSON]:
+        polling = kwargs.pop("polling", True)  # type: Union[bool, PollingMethod]
+        lro_delay = kwargs.pop("polling_interval", 0.5)
+
+        def operation() -> JSON:
+            return super(ConfidentialLedgerClientOperationsMixin, self).get_ledger_entry(
+                transaction_id, collection_id=collection_id, **kwargs
+            )
+
+        initial_response = operation()
+
+        if polling is True:
+            polling_method = cast(PollingMethod, StatePollingMethod(operation, "Ready", lro_delay))
+        elif polling is False:
+            polling_method = cast(PollingMethod, NoPolling())
+        else:
+            polling_method = polling
+
+        return LROPoller(self._client, initial_response, None, polling_method)
+
+    def begin_get_receipt(self, transaction_id: str, **kwargs: Any) -> LROPoller[JSON]:
+        polling = kwargs.pop("polling", True)  # type: Union[bool, PollingMethod]
+        lro_delay = kwargs.pop("polling_interval", 0.5)
+
+        def operation() -> JSON:
+            return super(ConfidentialLedgerClientOperationsMixin, self).get_receipt(
+                transaction_id=transaction_id, **kwargs
+            )
+
+        initial_response = operation()
+
+        if polling is True:
+            polling_method = cast(PollingMethod, StatePollingMethod(operation, "Ready", lro_delay))
+        elif polling is False:
+            polling_method = cast(PollingMethod, NoPolling())
+        else:
+            polling_method = polling
+
+        return LROPoller(self._client, initial_response, None, polling_method)
+
+    def begin_post_ledger_entry(
+        self,
+        entry: Union[JSON, IO],
+        *,
+        collection_id: Optional[str] = None,
+        **kwargs: Any,
+    ) -> LROPoller[JSON]:
+        """Writes a ledger entry and returns a poller to wait for it to be durably committed.
+
+        A collection id may optionally be specified.
+        """
+
+        # Pop arguments that are unexpected in the pipeline.
+        polling = kwargs.pop("polling", True)  # type: Union[bool, PollingMethod]
+        lro_delay = kwargs.pop("polling_interval", 0.5)
+
+        post_result = self.post_ledger_entry(entry, collection_id=collection_id, **kwargs)
+        transaction_id = post_result["transactionId"]
+
+        kwargs["polling"] = polling
+        kwargs["polling_interval"] = lro_delay
+        kwargs["_post_ledger_entry_response"] = post_result
```

Comment:
what is the hidden kwarg for?

Reply:
This is admittedly a bit hacky, but it allows `begin_post_ledger_entry` to directly return the poller from `begin_wait_for_commit` (because `begin_post_ledger_entry` is just a `post_ledger_entry` followed by polling on `begin_wait_for_commit`). `begin_wait_for_commit`'s poller would typically return the final transaction status for the transaction, but we'd like `begin_post_ledger_entry` to return the `post_ledger_entry` response instead, so we configure the poller to return the `post_ledger_entry` response if the `_post_ledger_entry_response` kwarg is provided.

Context from `begin_wait_for_commit` below:
![image](https://user-images.githubusercontent.com/9373058/176319592-0573413a-8b21-4210-86b4-c694ad25a5c2.png)

It's a little bit wacky but this minimizes code duplication, and I figured hidden kwargs would be reasonable to use.